virtual_rainforest/models/soil/soil_model.py

```diff
@@ -57,35 +66,21 @@ def __init__(
         data: Data,
         update_interval: timedelta64,
         start_time: datetime64,
-        no_layers: int,
         **kwargs: Any,
     ):
-        if no_layers < 1:
-            to_raise = InitialisationError(
-                "There has to be at least one soil layer in the soil model!"
-            )
-            LOGGER.error(to_raise)
-            raise to_raise
-
-        if no_layers != int(no_layers):
-            to_raise = InitialisationError(
-                "The number of soil layers must be an integer!"
-            )
-            LOGGER.error(to_raise)
-            raise to_raise
-
         super().__init__(data, update_interval, start_time, **kwargs)
-        self.no_layers = int(no_layers)
-        """The number of soil layers to be modelled."""
-        # Save variables names to be used by the __repr__
-        self._repr.append("no_layers")
 
         self.data
         """A Data instance providing access to the shared simulation data."""
         self.update_interval
         """The time interval between model updates."""
         self.next_update
         """The simulation time at which the model should next run the update method"""
+        self.carbon = SoilCarbonPools(data)
+        """Set of soil carbon pools used by the model"""
+
+        # Save variables names to be used by the __repr__
+        # self._repr.append("")
```

Comment:
What's happening here?

Reply:
I used to append things to the `_repr`, so that they would show up when `__repr__` is called. I'm not doing this for any variable any more, but I left it in commented out to remind myself. It probably should just go